src/GST.Fake.Authentication.JwtBearer/FakeJwtBearerHandler.cs

```diff
@@ -1,230 +1,230 @@
+        /// If it is not provided a default instance is supplied which does nothing when the methods are called.
+        /// </summary>
+        protected new JwtBearerEvents Events
+        {
+            get {
+                if(base.Events is JwtBearerEvents)
+                {
+                    return base.Events as JwtBearerEvents;
+                }
+                base.Events = new JwtBearerEvents();
+                return base.Events as JwtBearerEvents; }
+            set { base.Events = value; }
+        }
+
+        /// <summary>
+        /// Searches the 'Authorization' header for a 'Bearer' token.
+        /// </summary>
+        /// <returns></returns>
+        protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
+        {
+            string token = null;
+            try
+            {
+                // Give application opportunity to find from a different location, adjust, or reject token
+                var messageReceivedContext = new MessageReceivedContext(Context, Scheme, Options);
+
+                // event can set the token
+                await Events.MessageReceived(messageReceivedContext);
+                if (messageReceivedContext.Result != null)
+                {
+                    return messageReceivedContext.Result;
+                }
+
+                // If application retrieved token from somewhere else, use that.
+                token = messageReceivedContext.Token;
+
+                if (string.IsNullOrEmpty(token))
+                {
+                    string authorization = Request.Headers["Authorization"];
+
+                    // If no authorization header found, nothing to process further
+                    if (string.IsNullOrEmpty(authorization))
+                    {
+                        return AuthenticateResult.NoResult();
+                    }
+
+                    if (authorization.StartsWith("FakeBearer ", StringComparison.OrdinalIgnoreCase))
+                    {
+                        token = authorization.Substring("FakeBearer ".Length).Trim();
+                    }
+
+                    // If no token found, no further work possible
+                    if (string.IsNullOrEmpty(token))
+                    {
+                        return AuthenticateResult.NoResult();
+                    }
+                }
+
+                Dictionary<string, dynamic> tokenDecoded = JsonConvert.DeserializeObject<Dictionary<string, dynamic>>(token);
+
+                ClaimsIdentity id = new ClaimsIdentity("Identity.Application", Options.NameClaimType, Options.RoleClaimType);
+
+                foreach (var td in tokenDecoded)
+                {
+                    if (td.Key == "sub")
+                    {
+                        id.AddClaim(new Claim("sub", td.Value.ToString()));
+                        if (!tokenDecoded.Any(c => c.Key == "name"))
```

Comment:
Not totally sure if the `"name"` here should be changed to `Options.NameClaimType` too?